### DISCUSSION TITLE
How to test AES gcm-256bit AEAD encryption and decryption?

### DISCUSSION THREAD

**Discussion by goutham23** [openssl/openssl#21514] [Q&A]
How to test AES gcm-256bit AEAD encryption and decryption?

I'm seeing "enc: AEAD ciphers not supported"
Any other way?

**Comment by kulkarniamit** [comment]
* Related issue: #12220 
* [openssl-enc](https://www.openssl.org/docs/man3.0/man1/openssl-enc.html) documentation: 

  _"This command does not support authenticated encryption modes like CCM and GCM, and will not support such modes in the future. This is due to having to begin streaming output (e.g., to standard output when -out is not used) before the authentication tag could be validated. When this command is used in a pipeline, the receiving end will not be able to roll back upon authentication failure. The AEAD modes currently in common use also suffer from catastrophic failure of confidentiality and/or integrity upon reuse of key/iv/nonce, and since openssl enc places the entire burden of key/iv/nonce management upon the user, the risk of exposing AEAD modes is too great to allow. These key/iv/nonce management issues also affect other modes currently exposed in this command, but the failure modes are less extreme in these cases, and the functionality cannot be removed with a stable release branch. For bulk encryption of data, whether using authenticated encryption modes or other modes, [openssl-cms(1)](https://www.openssl.org/docs/man3.0/man1/openssl-cms.html) is recommended, as it provides a standard data format and performs the needed key/iv/nonce management."_

* [Demo program for encryption and decryption using aes-gcm](https://github.com/openssl/openssl/blob/master/demos/cipher/aesgcm.c)
